src/Config/RuntimeConfigLoader.cs

```diff
@@ -30,6 +31,39 @@ public abstract class RuntimeConfigLoader
     // state in place of using out params.
     public RuntimeConfig? RuntimeConfig;
 
+    public RuntimeConfigLoader(HotReloadEventHandler<HotReloadEventArgs>? handler = null, string? connectionString = null)
+    {
+        _changeToken = new DabChangeToken();
+        _handler = handler;
+        _connectionString = connectionString;
+    }
+
+    /// <summary>
+    /// Change token producer which returns an uncancelled/unsignalled change token.
+    /// </summary>
+    /// <returns>DabChangeToken</returns>
+#pragma warning disable CA1024 // Use properties where appropriate
+    public IChangeToken GetChangeToken()
+#pragma warning restore CA1024 // Use properties where appropriate
+    {
+        return _changeToken;
+    }
+
+    /// <summary>
+    /// Swaps out the old change token with a new change token and
+    /// signals that a change has occurred.
+    /// </summary>
+    /// <seealso cref="https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationProvider.cs">
+    /// Example usage of Interlocked.Exchange(...) to refresh change token.</seealso>
+    /// <seealso cref="https://learn.microsoft.com/en-us/dotnet/api/system.threading.interlocked.exchange?view=net-8.0">
+    /// Sets a variable to a specified value as an atomic operation.
+    /// </seealso>
+    private void RaiseChanged()
+    {
+        DabChangeToken previousToken = Interlocked.Exchange(ref _changeToken, new DabChangeToken());
```

Comment:
Why is a change token mechanism required as opposed to the event handlers that have already been added? What is different in signaling a change has occurred v/s an event has happened? Both seem to have subscribers...

Reply:
The objective was to trigger the `IOptionsMonitor` within the `JwtBearerHandler` to detect changes to authentication programmatically (as opposed to the out-of-box change detection for a bound configuration file such as `appsettings.json`, which we don't use). In order to alert the `IOptionsMonitor` that a change has occurred, I needed to register:
```csharp
services.AddSingleton<IOptionsChangeTokenSource<JwtBearerOptions>>(new JwtBearerOptionsChangeTokenSource(runtimeConfigProvider));
```
which is this resolved via DI in the `OptionsMonitor` class:
https://github.com/dotnet/runtime/blob/88397049d6aa2c8505bd11309ddd314169e3f73f/src/libraries/Microsoft.Extensions.Options/src/OptionsMonitor.cs#L31-L53

```csharp
        public OptionsMonitor(IOptionsFactory<TOptions> factory, IEnumerable<IOptionsChangeTokenSource<TOptions>> sources, IOptionsMonitorCache<TOptions> cache)
        {
            _factory = factory;
            _cache = cache;

            void RegisterSource(IOptionsChangeTokenSource<TOptions> source)
            {
                IDisposable registration = ChangeToken.OnChange(
                          source.GetChangeToken,
                          InvokeChanged,
                          source.Name);

                _registrations.Add(registration);
            }

            // The default DI container uses arrays under the covers. Take advantage of this knowledge
            // by checking for an array and enumerate over that, so we don't need to allocate an enumerator.
            if (sources is IOptionsChangeTokenSource<TOptions>[] sourcesArray)
            {
                foreach (IOptionsChangeTokenSource<TOptions> source in sourcesArray)
                {
                    RegisterSource(source);
                }
```

once our `runtimeconfigloader` detects a file change, it signals the provider that new `runtimeconfig` is available. Then the `RuntimeConfigProvider` (which is the service resolved via DI in most of our classes already) can signal the change token whose listener is the `OptionsMonitor` via registering the `IOptionsChangeTokenSource`. 

This mechanism was not as intrusive to our services' constructors:
1. No need to resolve `IOptionsMonitorCache<jwtBearerOptions>` in our `RuntimeConfigProvider` or `RuntimeConfigLoader` class constructors. Updating the `optionsmonitorcache` would be( i tested this) the manual way of changing `IOptionsMonitor<JwtBearerOptions>` used by the `JwtBearerHandler`, but this didn't work because of the next bullet point: 
    2. I made a judgement call that injecting `IoptionsmonitorCache<jwtBearerOptions>` into either of the two referenced classes seemed to leak implementation details of the provider/loader and wouldn't be straightforward without more time consuming and costly refactors:
        -  Because both the provider/loader are manually instantiated in `Startup::ConfigureServices` and don't have the opportunity to resolve services via DI.
        - Because we'd then need to make even more cascading changes in the test projects to accommodate setting up the `ioptionsmonitorcache` object to then be added to our mock runtimeconfigprovider/loader objects.

The event-handlers we have recently added don't fit this specific use case of updating the underlying authentication configuration. In addition to the above design decisions, I didn't identify a solution where we could subscribe the `JwtBearerHandler`'s `OptionsMonitor`  to an eventhandler signal to notify that a change occurred. I pursued using the built-in mechanism of change detection for optionsmonitor provided to us via `IOptionsChangeTokenSource`.

Reply:
adding the above response to the PR description